src/utils/session.ts

```diff
@@ -22,3 +22,26 @@ export function getPresenterAccountDetails(req: Request): Details {
 export function setPresenterAccountDetails(req: Request, details: Details) {
     req.session?.setExtraData(PRESENTER_ACCOUNT_SESSION_KEY, details);
 }
+
+export function generateUserDetails(req: Request): Details{
+    const signinInfo = req.session?.data?.signin_info;
+    const details = {} as Details;
+
+    details.email = signinInfo?.user_profile?.email as string;
+    details.userId = signinInfo?.user_profile?.id as string;
```

Comment:
If the details object is moved to the bottom of the function it can also be initialised as an object literal.